src/desktop/DesktopUtils.ts

```diff
@@ -101,50 +88,62 @@ export class DesktopUtils {
 	 *
 	 * @returns {Promise<boolean>} whether the app was successful in getting the lock
 	 */
-	makeSingleInstance(): Promise<boolean> {
-		const lockfilePath = this.getLockFilePath()
-		// first, put down a file in temp that contains our version.
-		// will overwrite if it already exists.
-		// errors are ignored and we fall back to a version agnostic single instance lock.
-		return this.fs.promises
-			.writeFile(lockfilePath, this.electron.app.getVersion(), "utf8")
-			.catch(noOp)
-			.then(() => {
-				// try to get the lock, if there's already an instance running,
-				// give the other instance time to see if it wants to release the lock.
-				// if it changes the version back, it was a different version and
-				// will terminate itself.
-				return this.electron.app.requestSingleInstanceLock()
-					? Promise.resolve(true)
-					: delay(1500)
-							.then(() => this.singleInstanceLockOverridden())
-							.then((canStay) => {
-								if (canStay) {
-									this.electron.app.requestSingleInstanceLock()
-								} else {
-									this.electron.app.quit()
-								}
+	async makeSingleInstance(): Promise<boolean> {
+		const isOnlyInstance = await this.tfs.acquireSingleInstanceLock()
+		if (isOnlyInstance) {
+			return true
+		}
+		// the other instance will decide if it's going to terminate and override
+		// the lock again in this time.
+		await delay(1500)
+		const otherInstanceWillTerminate = await this.tfs.singleInstanceLockOverridden()
+		if (otherInstanceWillTerminate) {
+			this.electron.app.requestSingleInstanceLock()
+		} else {
+			this.electron.app.quit()
+		}
+
+		return otherInstanceWillTerminate
+	}
 
-								return canStay
-							})
-			})
+	/** after we receive notification about another app instance being started, we need to decide
```

Comment:
sorry about being so annoying about the comments but jsdoc comment is either a single line or starts and ends with empty lines (`/**` and `*/`)

Reply:
can´t we add this to eslint ~nig

Reply:
there's https://github.com/hosseinmd/prettier-plugin-jsdoc but that requires prettier 3

Reply:
it doesn't?
https://github.com/hosseinmd/prettier-plugin-jsdoc?tab=readme-ov-file#supported-prettier-version

and also we could upgrade prettier anyway